sui-execution/latest/sui-adapter/src/static_programmable_transactions/linkage/analysis.rs

```diff
@@ -17,65 +16,99 @@ use sui_types::{
     transaction as P,
 };
 
-pub trait LinkageAnalysis {
-    fn compute_call_linkage(
+#[derive(Debug)]
+pub struct LinkageAnalyzer {
+    internal: ResolutionConfig,
+}
+
+impl LinkageAnalyzer {
+    pub fn new<Mode: ExecutionMode>(
```

Comment:
Should this be named `legacy_linkage` instead of `new`?

Reply:
I went back-and-forth on it, but I don't think we should, and this should encapsulate legacy vs non-legacy analysis internally within it and not expose it, and the selection of which type of analysis is selected is done by the protocol config passed in.